scc/provider/type_system_mapping.go

```diff
@@ -47,11 +53,44 @@ type SystemMapping struct {
 	EnabledResourcesCount types.Int64  `tfsdk:"enabled_resources_count"`
 	Description           types.String `tfsdk:"description"`
 	SAPRouter             types.String `tfsdk:"sap_router"`
+	SNCPartnerName        types.String `tfsdk:"snc_partner_name"`
+	AllowedClients        types.List   `tfsdk:"allowed_clients"`
+	BlacklistedUsers      types.List   `tfsdk:"blacklisted_users"`
 }
 
-func SystemMappingsValueFrom(ctx context.Context, plan SystemMappingsConfig, value apiobjects.SystemMappings) (SystemMappingsConfig, error) {
+type SystemMappingBlacklistedUsersData struct {
+	Client types.String `tfsdk:"client"`
+	User   types.String `tfsdk:"user"`
+}
+
+var SystemMappingBlacklistedUsersType = types.ObjectType{
+	AttrTypes: map[string]attr.Type{
+		"client": types.StringType,
+		"user":   types.StringType,
+	},
+}
+
+func SystemMappingsValueFrom(ctx context.Context, plan SystemMappingsConfig, value apiobjects.SystemMappings) (SystemMappingsConfig, diag.Diagnostics) {
 	system_mappings := []SystemMapping{}
 	for _, mapping := range value.SystemMappings {
+		blacklistedUsersValue := []SystemMappingBlacklistedUsersData{}
+		for _, user := range mapping.BlacklistedUsers {
+			bl := SystemMappingBlacklistedUsersData{
+				Client: types.StringValue(user.Client),
+				User:   types.StringValue(user.User),
+			}
+			blacklistedUsersValue = append(blacklistedUsersValue, bl)
+		}
+		blacklistedUsers, err := types.ListValueFrom(ctx, SystemMappingBlacklistedUsersType, blacklistedUsersValue)
```

Comment:
Although this might be a larger rework: the Terraform functions usually return a `diags` object, so we should use that consistently. Using `err` might be misleading or even lead to wrong assignments. 

I would propose to rename this consistently. Probably in a new PR for all resources -> I will open a follow-up issue for this.